database/sinkdb/op.go

```diff
@@ -49,6 +49,22 @@ func IfNotExists(key string) Op {
 	}
 }
 
+// IfMatch encodes a conditional to make an instruction
+// successful only if the stored version matches v.
+// It checks the key returned by method Key on v.
+//
+// If v.Exists() is false,
+// IfMatch(v) is equivalent to IfNotExists(v.Key()).
+func IfMatch(v Version) Op {
```

Comment:
(I started with IfMatch because that's what HTTP calls the equivalent header field, but that's not a very good reason.)